### ISSUE TITLE
[ILM] Handle the case when the frozen phase repository is different to the repository used in cold

### ISSUE THREAD

**Issue by andreidan** [elastic/elasticsearch#90792]
### Description

The `searchable_snapshot` action saves the configured snapshot repository in the ILM state.
If a policy with both `cold` and `frozen` phases use different snapshot repository for the fully mounted (cold) and partially mounted (frozen) indices, ILM will fail in the `frozen` phase with:
`searchable snapshot indices may be converted only within the same repository`

Maybe a bit unexpected, but note that this can happen if a policy's frozen phase is **updated** to mount indices in a different repository after indices have entered the frozen phase (this is due to the fact that we'll update the frozen cached phase to point to the new repository, whilst the execution state will continue to hold the snapshot repository configured in the cold phase)

We should handle this better as the workaround is not simple because we need the partial mount to happen based on a snapshot that contains the **original** index which needs to exist in the repository name that's stored in the index lifecycle execution state (it feels easier to simply delete the mounted index, restore the original index from a snapshot, and then attach a policy that only contains the `frozen` and `delete` phases to partially mount the index)

**Comment by andreidan** [comment]
Discussed and decided that for now we'll only validate that one repository is used per ILM policy at creation time.

There are various scenarios around updating an ILM policy (updating the `searchable_snapshot` action in a phase previous or after the current execution phase for some indices) which would entail us to support the full scenario of possibly restoring the original index in the cluster, re-taking a snapshot in the new repository and mounting the index (fully or partially). 

We've decided that for now we'll only validate the ILM policy at creation time.

**Comment by HiDAl** [comment]
Hey @andreidan 

I wanted to work on the issue but I stumbled upon that apparently we're already validating this here: https://github.com/elastic/elasticsearch/blob/main/x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ilm/TimeseriesLifecycleType.java#L380-L398

Am I correct or misunderstood the issue??

**Comment by andreidan** [comment]
@HiDAl good catch. That covers the policy creation time validation and we seem to validate the update of policies too since https://github.com/elastic/elasticsearch/pull/88702 (the update case still has loopholes - iterative, repeated, updates could still yield indices that have been managed by `policy_1` and mounted in `repostitory_1` to end up being managed by `policy_1` configured to mount in `repository_2` - but as we decided to handle the basic cases this is fine for now)

I think the existing behaviour is enough for now.
